document/cloud-client/batch_parse_form.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START documentai_batch_parse_form]
+from google.cloud import documentai_v1beta2 as documentai
+from google.cloud import storage
+import re
+
+
+def batch_parse_form(
+        project_id='YOUR_PROJECT_ID',
+        input_uri='gs://cloud-samples-data/documentai/form.pdf',
+        destination_uri='DESTINATION_URI'):
+    """Parse a form"""
+
+    client = documentai.DocumentUnderstandingServiceClient()
+
+    gcs_source = documentai.types.GcsSource(uri=input_uri)
+
+    # mime_type can be application/pdf, image/tiff,
+    # and image/gif, or application/json
+    input_config = documentai.types.InputConfig(
+        gcs_source=gcs_source, mime_type='application/pdf')
+
+    # where to write results
+    output_config = documentai.types.OutputConfig(
+        gcs_destination=documentai.types.GcsDestination(
+            uri=destination_uri),
+        pages_per_shard=1  # Map one doc page to one output page
+    )
+
+    # Improve form parsing results by providing key-value pair hints.
+    # For each key hint, key is text that is likely to appear in the
+    # document as a form field name (i.e. "DOB").
+    # Value types are optional, but can be one or more of:
+    # ADDRESS, LOCATION, ORGANIZATION, PERSON, PHONE_NUMBER, ID,
+    # NUMBER, EMAIL, PRICE, TERMS, DATE, NAME
+    key_value_pair_hints = [
+        documentai.types.KeyValuePairHint(
+            key='Emergency Contact',
+            value_types=['NAME']),
+        documentai.types.KeyValuePairHint(
+            key='Referred By')
+    ]
+
+    # Setting enabled=True enables form extraction
+    form_extraction_params = documentai.types.FormExtractionParams(
+        enabled=True, key_value_pair_hints=key_value_pair_hints)
+
+    # For now, location must be us-central1
+    parent = "projects/{}/locations/us-central1".format(project_id)
+    request = documentai.types.ProcessDocumentRequest(
+        input_config=input_config,
+        output_config=output_config,
+        form_extraction_params=form_extraction_params)
+
+    requests = documentai.types.BatchProcessDocumentsRequest(
+        parent=parent, requests=[request]
```

Comment:
We've had feedback that people get confused by this sample code for batch requests that puts the request object inside an array. Basically there is confusion about how to send multiple requests. 

Mind adding some comments like:
1. Build a single request for one filed
2. Now put that one request object inside the batch request, you can send multiple requests but you'll have to create them yourself. 

I updated this Java sample below if you want to see sorta what I mean. 


Example: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/vision/cloud-client/src/main/java/com/example/vision/AsyncBatchAnnotateImages.java

Reply:
Done